topic-controller/src/main/java/io/strimzi/controller/topic/Controller.java

```diff
@@ -47,47 +46,63 @@
     private TopicStore topicStore;
     private final InFlight inFlight;
 
-    class ErrorEvent implements Handler<Void> {
+    enum EventType {
+        INFO("Info"),
+        WARNING("Warning");
+        final String name;
+        EventType(String name) {
+            this.name = name;
+        }
+    }
 
+    class Event implements Handler<Void> {
+        private final EventType eventType;
         private final String message;
         private final HasMetadata involvedObject;
         private final Handler<AsyncResult<Void>> handler;
 
-        public ErrorEvent(ControllerException exception, Handler<AsyncResult<Void>> handler) {
+        public Event(ControllerException exception, Handler<AsyncResult<Void>> handler) {
             this.involvedObject = exception.getInvolvedObject();
             this.message = exception.getMessage();
             this.handler = handler;
+            this.eventType = EventType.WARNING;
         }
 
-        public ErrorEvent(HasMetadata involvedObject, String message, Handler<AsyncResult<Void>> handler) {
+        public Event(HasMetadata involvedObject, String message, EventType eventType, Handler<AsyncResult<Void>> handler) {
             this.involvedObject = involvedObject;
             this.message = message;
             this.handler = handler;
+            this.eventType = eventType;
         }
 
         @Override
         public void handle(Void v) {
-            String myHost = "";
             EventBuilder evtb = new EventBuilder().withApiVersion("v1");
             if (involvedObject != null) {
                 evtb.withNewInvolvedObject()
-                        .withKind(involvedObject.getKind())
+                        .withKind("ConfigMap")
                         .withName(involvedObject.getMetadata().getName())
                         .withApiVersion(involvedObject.getApiVersion())
                         .withNamespace(involvedObject.getMetadata().getNamespace())
                         .withUid(involvedObject.getMetadata().getUid())
                         .endInvolvedObject();
             }
-            evtb.withType("Warning")
-                    .withMessage(this.getClass().getSimpleName() + " failed: " + message)
-                    //.withReason("")
+            evtb.withType(eventType.name)
+                    .withMessage(message)
+                    .withNewMetadata().withLabels(cmPredicate.labels()).withGenerateName("topic-controller").endMetadata()
```

Comment:
Because as @scholzj pointed out, the Event API (so this `Event` class) could be common with cluster-controller (and other component which will come), maybe it's possible to avoid hard-conding "topic-controller" here enabling this class to be used elsewhere ?

Reply:
I assume the topic-controller won't be a maven dependency of the cluster controller and that when we're ready to use common stuff we will move it into some `strimzi-common` maven artifact. imho that would be the time to remove such hard coding.

Reply:
There will be anyway some more work needed to have this as common code. So maybe we don't have to deal with this right now.